pymeasure/instruments/common_base.py

```diff
@@ -297,12 +297,14 @@ def values(self, command, separator=',', cast=float, preprocess_reply=None, maxs
         values from the result.
 
         :param command: SCPI command to be sent to the instrument.
-        :param separator: A separator character to split the string into a list.
-        :param cast: A type to cast the result.
-        :param preprocess_reply: optional callable used to preprocess values
-            received from the instrument. The callable returns the processed
-            string.
-        :param maxsplit: At most `maxsplit` splits are done. -1 (default) indicates no limit.
+        :param separator: A separator character to split the string returned by
+            the device into a list.
+        :param cast: A type to cast each element of the splitted string.
+        :param preprocess_reply: Optional callable used to preprocess the string
+            received from the instrument, before splitting it.
+            The callable returns the processed string.
+        :param maxsplit: The string returned by the device is splitted at most `maxsplit` times.
+            -1 (default) indicates no limit.
```

Comment:
Very nice clarifications on the argument descriptions!
To improve this even more, what do you think about adjusting their order in the order they become relevant/applied? I.e. `preprocess_reply`, `separator`, `maxsplit`, `cast`? 
It's a little thing, but could improve clarity/understanding when studying the documentation/docstring?

Reply:
I thought about it, but I kept them in order of the signature.
Probably we can change the signature, but it can be dangerous (if people use args and not kwargs).

Reply:
We could make them keyword-only then any positional use would be an error?
I don't think I have ever seen property factories where those args were positional. For `values`, I'm not sure.

Reply:
I was thinking about values.

That's the problem with backwards compatibility...

We can order the docs differently than the args.

With "*args" we can prevent normal arguments (or how else do you enforce kwargs?)

Reply:
> `func(posarg1, posarg2, *, kwarg1=bla, kwarg2=blubb), IIRC.`

That is, what I meant, with "*args".

I did a small test, the star works as you suggest (without assigning it to some list).

Reply:
Yeah, this is https://python-3-for-scientists.readthedocs.io/en/latest/python3_advanced.html#keyword-only-arguments.
What we would be making use of:
> It also becomes possible to reorder keyword-only arguments in a function signature, since keyword-only arguments are only accessible via the name of the argument, not its position in the function signature.

Not sure if the small benefit (better order in the signature) is worth it.

Reply:
We can do the documentation in logical order and only keep the `values` signature "disordered" (as-is), while `control` has the parameters in logical order (in sync with the documentation)